packages/core/src/modules/x509/X509Certificate.ts

```diff
@@ -57,22 +57,18 @@ export enum X509ExtendedKeyUsage {
 export type X509CertificateOptions = {
   publicKey: Key
   privateKey?: Uint8Array
-  extensions?: Array<x509.Extension>
-  rawCertificate: Uint8Array
+  x509Certificate: x509.X509Certificate
```

Comment:
I don't think it should be a required property right? Makes it very strict, we can build the x509 cert from the options, right?

Reply:
Ah I see its just a change from the raw certificate.

Maybe we should allow to pass in the raw or the x509.X509Certificate? Otherwise the user would need to install the @peculiar lib

Reply:
So i'll make the constructor private, and you can use the static members (fromEncodedCertificate, fromRawCertificate)